pykern/pkconfig.py

```diff
@@ -668,10 +679,13 @@ def _coalesce_values():
     values[CHANNEL_ATTR] = channel
     _raw_values = values
     _parsed_values = dict(((_Key([k]), v) for k, v in env.items()))
-    cfg = init(
+    a = PKDict(
         _caller_module=sys.modules[__name__],
         channel=Required(str, "which (stage) function returns config"),
     )
+    cfg = init(**a)
+    a.dev_mode = (channel_in("dev"), bool, "controls development features")
+    cfg = init(**a)
```

Comment:
dont understand why a.dev_mode is being set in between two calls to `init()`

Reply:
The first call sets `cfg.channel` from the specification in `a`. `channel_in` is then used to initialize `a.dev_mode`, then it is passed to `init`, which sets cfg.dev_mode.